src/js/select2/selection/search.js

```diff
@@ -54,6 +54,10 @@ define([
       self.$search.trigger('focus');
     });
 
+    container.on('results:speak', function (params) {
```

Comment:
Instead of using a custom event here, you can hook into `results:focus` and use `params.element.id` to get the `id` of the element.